### DISCUSSION TITLE
Ecolink Freeze Flood Notification Code Error?

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js#3162] [Q&A] [answered]
Since I'm new to zwave-js, I wanted to raise this here before assuming it was an issue. I have an Ecolink Flood/Freeze Sensor and I have a Binary Sensor (113-0-Water Alarm-Sensor Status in CP) and a second binary sensor (which doesn't appear in the zwavejs2mqtt CP, but does in Home Assistant), which tracks the 113-0 (NOTE: this is NOT the 48-0-Water binary sensor in the CP). 

What is happening is that when the sensor is triggered by water, the 48-0-Water binary sensor goes TRUE, and the 113-0-Water Alarm-Sensor Status goes to "Water Leak Detected" - both as they should. When the Water is removed, the 48-0 value goes to FALSE, but the 113-0 remains at "Water Leak Detected". This is true with that second BS in HA that tracks 113-0, which is set to On in HA and stays On even after the water is removed.

I'm not sure, but I believe both of these latter values (that don't change back) are the Notification Type "Water (0x05)" (consistent with what I'm seeing in the logs). I can see the 0x02 value Set when Water is Detected initially. However, when the water is removed, there is no entry to unSet it. Per the docs, I think we should see a 0x04 "Water No Longer Detected" being sent/received when water is removed.

I don't know if this is a bug in the driver, or an error in the device config files used by it, or if the device is ACTUALLY NOT sending a 0x04 as the manual indicates. I've attached the manual, and the values I'm talking about are in the Notification CC Table on Pg 6.

Also, In the attached driver (and zw2m) logs, look at Nodes 42, 62, and 63, all of which I tested by wetting, and all responded identically as described above. In the logs, you will see that on the first wetting, there is a Notification CC set from 0 to 2, consistent with the manual. On following tests, the Notification CC set is from 2 to 2 upon wetting. It is never being reset after water is removed (0x04). I assume it should. The 48-0 sensor is working correctly, and the logs show it going to TRUE and then back to FALSE.

EDIT: Actually, looking at the zw2m logs, I can see a 0x04 for Notification CC Type 0x05 (Water) where it should be, as shown here.

```
2021-08-07 00:17:33.021 INFO ZWAVE: Node 63 CC Notification {
  type: 5,
  event: 4,
  label: 'Water Alarm',
  eventLabel: 'Water level dropped',
  parameters: undefined
}
```

It corresponds to the following in the driver log.

```
2021-08-07T08:36:24.255Z DRIVER « [Node 063] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Water Alarm
                                      notification status: 255
                                      notification event:  Water level dropped
```
I never see the "Water Level Dropped", just the "Water Detected".

Please advise, if possible. Thanks.
[zwavejs2mqtt-store.zip](https://github.com/zwave-js/node-zwave-js/files/6948894/zwavejs2mqtt-store.zip)
[Z-Wave Plus Wireless Flood Sensor with a Probe FLF-ZWAVE5_V5.pdf](https://github.com/zwave-js/node-zwave-js/files/6948895/Z-Wave.Plus.Wireless.Flood.Sensor.with.a.Probe.FLF-ZWAVE5_V5.pdf)

**Comment by kpine** [comment]
See this discussion in an HA issue: https://github.com/home-assistant/core/issues/51691#issuecomment-858300763

The easiest solution is to just ignore the Notification-based binary sensor and use the Binary Sensor-based binary sensor (after enabling it) that is working. Other solutions would depend on custom behavior for this device in node-zwave-js.

**Reply by GrizzlyAK** [reply]
Well, CRAP. I didn't think to search the HA/Core Github, but this is exactly the same issue as discussed there. I have a much better understanding about what's going on now after reading that discussion. Thanks for that well-written response.

But, is HA creating the _additional_ binary.sensor that tracks, or reflects the value stored in, the Notification CC Water (0x05) Sensor, since BOTH of these values take on and MAINTAIN their positive values once triggered, and never reset? Because I don't see that additional sensor in the zw2m CP. This binary sensor is DIFFERENT than the one created for the Sensor Binary CC (which works as it should).

Also, I'm working with Ecolink currently testing their new siren+chime, so I will pass along that they are not implementing the Notification CC V5 correctly in regards to this issue.

**Reply by kpine** [reply]
HA creates binary sensors based on the possible states in a notification state variable (in addition to the state variable which is a disabled-by-default sensor). zwavejs2mqtt will only show the state variable. The value 0x4 triggers an "on" value for that generated binary sensor. It would be set to "off" if the device properly reset it with the Idle (0x0) value, or some other value (for multi-state variables).

**Reply by GrizzlyAK** [reply]
Okay! That explains what I was seeing regarding the _additional_ entities not appearing in the CP. Thanks!

**Comment by AlCalzone** [comment, marked as answer]
> a second binary sensor ..., which tracks the 113-0

I'm not sure what that means. Value IDs consist of command class (113), endpoint (0) and at least a property, sometimes also a property key.

> In the logs, you will see that on the first wetting, there is a Notification CC set from 0 to 2, consistent with the manual. On following tests, the Notification CC set is from 2 to 2 upon wetting. It is never being reset after water is removed (0x04). I assume it should. 

I think this is one of the cases where the Z-Wave specifications need to be improved. **Water Leak Detected (2)** is defined in the specs as a state (the value has a meaning after the notification is received). **Water Level Dropped (4)** is defined as an event (that has a meaning only the instant the notification is received). I don't see any indication that those are related to each other. **Water Level Dropped** could just mean that you have a container where the water level went down, e.g. from half full to quarter full.

To reset a notification **state**, Z-Wave has a special **idle (0)** notification, which the device should be using instead. Since you have a working binary sensor, I agree with @kpine - it isn't worth the developer time to implement a workaround.

**Reply by GrizzlyAK** [reply]
> I'm not sure what that means. 

Please see my response to kpine above. This must be a sensor that HA creates to track the Notification Sensor, since I don't see it in the zw2m CP. 

> Water Level Dropped (4) is defined as an event (that has a meaning only the instant the notification is received).

Looking at the image in the HA/Core GH post kpine referenced, I see what you mean now. It appears Ecolink was a little confused when choosing which value to send, an. Although 0x04 is "Water Level Dropped", they reference it as "Water No Longer Detected", which, as you pointed out, should have been 0x00 "Idle".

Now that I understand WHAT is going on, I can choose which sensors to use for my purposes. Thanks to both of you.
